sourcegraph/browser-extensions
# feat(cxp): add options page with CXP extensions list

![2018-08-16 19 22 04](https://user-images.githubusercontent.com/1387653/44244670-de7b1580-a189-11e8-97fd-3893d5b811cc.gif)

This PR changes:

- Adds an options page under the "Enable CXP" feature flag (right click on the icon, select Options)

Testing plan:

- Check "Enable CXP"
- Open the options page

- You should see CXP extensions from your Primary Sourcegraph instance
- Toggling extensions on/off should change their associated value in the client settings at the bottom of the page to true/false
- Removing an extension should remove it from the client settings
- Adding an extension should add it to client settings

I have tested on:


cc @KattMingMing Take note of `cxp.html` and `cxp.tsx` - these will get replaced by or embedded into your new options UI.

TODO

- [x] Don't show user/org/site options because this is for client settings only
- ~[ ] Link to the upstream extension registry when the user clicks on an extension~ easier accomplished by changing the GraphQL API of Sourcegraph
